plugins/vpc-tunnel/plugin/commands.go

```diff
@@ -0,0 +1,360 @@
+				tapBridge := &netlink.Bridge{LinkAttrs: la}
+				log.Infof("Deleting tap bridge: %v.", tapBridgeName)
+				err = netlink.LinkDel(tapBridge)
+				if err != nil {
+					log.Errorf("Failed to delete tap bridge: %v.", err)
+					errs = multierror.Append(err, errs)
+				}
+			}
+
+			return errs
+		})
+		if err != nil {
+			return err
+		}
+	} else {
+		// Log and ignore the failure. DEL can be called multiple times and thus must be idempotent.
+		log.Errorf("Failed to find netns %s, ignoring: %v.", args.Netns, err)
+	}
+
+	return nil
+}
+
+// createGeneveLink creates the GENEVE interface and moves it to desired netns.
+// If it already exists and is the primary interface, it is reset.
+func (plugin *Plugin) createGeneveLink(
+	geneve *eni.Geneve,
+	netConfig *tunnelConfig.NetConfig,
+	ns netns.NetNS) error {
+	err := geneve.AttachToLink()
+	if err != nil {
+		if err.Error() == "device or resource busy" {
+			// If the geneve link already exists, it may have been created in a previous invocation
+			// of this plugin. Look for it in the target network namespace and reset it.
+			err = ns.Run(func() error {
+				err := geneve.ENI.AttachToLink()
+				if err != nil {
+					return err
+				}
+
+				// IP address and routing rules are only put in place for the interface
+				// only if it was the primary interface. So for non primary interface,
+				// we do not have to delete the same while reseting.
+				if geneve.Primary {
+					if err = plugin.resetGeneveLink(geneve, netConfig); err != nil {
+						log.Errorf("Failed to reset geneve link: %v.", err)
+						return err
+					}
+				}
+				return nil
+			})
+		}
+		if err != nil {
+			log.Errorf("Failed to attach geneve interface %s: %v.", geneve.ENI.GetLinkName(), err)
+			return err
+		}
+	} else {
+		// Move GENEVE link to the network namespace.
+		log.Infof("Moving geneve link %s to netns %s.", geneve, ns.GetPath())
+		err = geneve.SetNetNS(ns)
+		if err != nil {
+			log.Errorf("Failed to move geneve link: %v.", err)
+			return err
+		}
+	}
+
+	return err
+}
+
+// resetGeneveLink deletes all IP addresses, routes and ARP table entries related to the GENEVE
+// interface thus resetting it.
+func (plugin *Plugin) resetGeneveLink(geneve *eni.Geneve, netConfig *tunnelConfig.NetConfig) error {
+	hwAddr, err := net.ParseMAC(defaultGatewayMacAddr)
+	if err != nil {
+		return err
+	}
+
+	neigh := &netlink.Neigh{
+		IP:           netConfig.GatewayIPAddress,
+		HardwareAddr: hwAddr,
+		LinkIndex:    geneve.GetLinkIndex(),
+	}
+	log.Infof("Deleting neighbour %+v.", neigh)
+	if err = netlink.NeighDel(neigh); err != nil {
+		return err
+	}
+
+	route := &netlink.Route{
+		Gw:        netConfig.GatewayIPAddress,
+		LinkIndex: geneve.GetLinkIndex(),
+	}
+	log.Infof("Deleting default IP route %+v.", route)
+	err = netlink.RouteDel(route)
+	if err != nil {
+		// Trying to delete a route which does not exist will yield a 'no such process error'.
+		if err.Error() != "no such process" {
+			return err
```

Comment:
If the route doesn't exist, why do we treat it as an error?

Reply:
`no such process` happens if the route does not exist. So here we are treating all other messages as errors.